sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -37,14 +38,33 @@
 
 def _get_arg(typ, index):
   """Returns the index-th argument to the given type."""
-  return typ.__args__[index]
+  try:
+    return typ.__args__[index]
+  except AttributeError:
+    if (3, 0, 0) <= sys.version_info[0:3] < (3, 5, 3):
+      # On Python versions < 3.5.3, the Tuple and Union type from typing do
+      # not have an __args__ attribute, but a __tuple_params__, and a
+      # __union_params__ and __union_set_params__ argument respectively.
+      args = next(value for key, value in typ.__dict__.items()
```

Comment:
This looks correct, but is hard to read. I suggest, we rewrite this to be more explicit here:

```
if has_attr(typ, '__tuple_params__') :
  args = typ.__tuple_params__
elif has_attr(typ, '__union_params__'):
  args = typ.__tuple_params__
else:
  assert False, "Only Union or Tuple metatypes are expected here."
return args[index]
```
We can do a similar change below.

Also R: @charlesccychen to take another look on this change, thank you.

FYI, 3.5.2 seems to use this revision of typing.py https://github.com/python/cpython/blob/3b557991d4a7626cf12baf2277bc87acbc439744/Lib/typing.py, changed in https://github.com/python/cpython/commit/5fc25a873cfdec27e46f71e62c9b65df5667c1b4#diff-015978a768b517a38abbc0ecdea87f5a.

Reply:
I sent out https://github.com/apache/beam/pull/7902 to implement @tvalentyn's suggestion

Reply:
I tried to make this check generic, not knowing if there were any other types following these same conventions. However, looking at the linked revision, it seems limited to `Tuple` and `Union`, so I agree that it would be better to clean this up.

Thanks!